### PR TITLE
Doing some TODOs

### PR DESCRIPTION
**Pull request by element-of-surprise**
Rename of some methods
Switching out string/int representations of time for a new time.Unix object in a json/time package.
Some minor changes to json to help with struct fields that are not pointers.

### CODE REVIEW

File: apps/confidential/confidential.go [hunk 1]
```diff
@@ -135,12 +135,12 @@ type Credential struct {
 	key  crypto.PrivateKey
 }
 
-// toMSALBASE returns the accesstokens.Credential that is used internally. The current structure of the
+// toInternal returns the accesstokens.Credential that is used internally. The current structure of the
 // code requires that client.go, requests.go and confidential.go share a credential type without
 // having import recursion. That requires the type used between is in a shared package. Therefore
 // we have this.
 // TODO(jdoak): change method name.
```
Comment (keegan-caruso) [inline apps/confidential/confidential.go:142]: remove?

File: apps/internal/base/internal/storage/items.go [hunk 2]
```diff
@@ -74,28 +74,28 @@ type AccessToken struct {
 	Secret         string `json:"secret,omitempty"`
 	Scopes         string `json:"target,omitempty"`
 	// TODO(jdoak): This should probably be a wrapper around time.Time that json marshals.
```
Comment (keegan-caruso) [inline apps/internal/base/internal/storage/items.go:76]: remove these three

File: apps/internal/json/types/time/time.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package time
```
Comment (keegan-caruso) [inline apps/internal/json/types/time/time.go:1]: copyright

File: apps/internal/json/types/time/time.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Unix provides a type that can marshal and unmarshal a string representation
+// of the unix epoch into a time.Time object.
+type Unix struct {
```
Comment (keegan-caruso) [inline apps/internal/json/types/time/time.go:17]: think this types warrants a line in json/design.md?

File: apps/internal/json/json.go [hunk 5]
```diff
@@ -137,7 +141,6 @@ func callMarshalJSON(v reflect.Value) ([]byte, error) {
 		if v.CanAddr() {
 			v = v.Addr()
 		}
-		panic(fmt.Sprintf("callMarshalJSON called on type %T that does not have MarshalJSON defined", v.Interface()))
```
Comment (keegan-caruso) [inline apps/internal/json/json.go:140]: originally a bug or a change in thinking on this?

Comment (element-of-surprise) [reply apps/internal/json/json.go:140]: This was a bug.  The if condition didn't even matter there.

File: apps/internal/json/types/time/time.go (hunk 4 above)
Comment (element-of-surprise) [reply apps/internal/json/types/time/time.go:17]: Nahhh.

File: apps/internal/base/internal/storage/items_test.go [hunk 6]
```diff
@@ -365,6 +365,7 @@ func TestContractMarshalJSON(t *testing.T) {
 	if err != nil {
 		t.Fatalf("TestContractMarshalJSON(marshal): got err == %s, want err == nil", err)
 	}
+	log.Println(string(b))
```
Comment (keegan-caruso) [inline apps/internal/base/internal/storage/items_test.go:368]: debug output?

Comment (element-of-surprise) [reply apps/internal/base/internal/storage/items_test.go]: yep, removed.

**Review by keegan-caruso — APPROVED** [review]
LGTM

**Pull request merged**